packages/babel-generator/src/printer.ts

```diff
@@ -674,13 +664,22 @@ class Printer {
     this._insideAux = node.loc == undefined;
     this._maybeAddAuxComment(this._insideAux && !oldInAux);
 
-    const shouldPrintParens =
+    let shouldPrintParens =
       forceParens ||
       (format.retainFunctionParens &&
         nodeType === "FunctionExpression" &&
         node.extra?.parenthesized) ||
       needsParens(node, parent, this._printStack);
 
+    if (
+      !shouldPrintParens &&
+      parent?.type === "MemberExpression" &&
```

Comment:
I think we don't have to limit the parent type. If there are leading comments and the node is parenthesized, it suggests that users intend to keep comments within the parentheses. Examples are
```js
(/* @__PURE__ */ f())()
(/* @__PURE__ */ f())?.bar
(/* @__PURE__ */ f())`bar`
```

Reply:
```
    -       var c = /*#__PURE__*/function () {
    +       var c = ( /*#__PURE__*/function () {
```

```
    -       setTimeout( /*#__PURE__*/function () {
    +       setTimeout(( /*#__PURE__*/function () {
```
Unfortunately there are some unexpected situations.